Hackney.Core/Hackney.Core.Authorization/AuthorizeEndpointByIpWhitelist.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using System.Collections.Generic;
+using Hackney.Core.Authorization.Exceptions;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.Logging;
+
+namespace Hackney.Core.Authorization
+{
+    /// <summary>
+    /// Represents a class that authorizes an endpoint based on an IP whitelist.
+    /// </summary>
+    public class AuthorizeEndpointByIpWhitelist : TypeFilterAttribute
+    {
+        /// <summary>
+        /// Attribute that wraps the <see cref="Hackney.Core.Authorization.AuthorizeEndpointByIpWhitelist"/> to authorize an endpoint based on an IP whitelist.
+        /// </summary>
+        /// <param name="ipWhitelistVariableName">The name of the IP whitelist variable.</param>
+        public AuthorizeEndpointByIpWhitelist(string ipWhitelistVariableName) : base(typeof(IpWhitelistFilter))
+        {
+            Arguments = new object[] { ipWhitelistVariableName };
+        }
+    }
+
+    /// <summary>
+    /// Represents a class that filters requests based on an IP whitelist.
+    /// </summary>
+    public class IpWhitelistFilter : IAuthorizationFilter
+    {
+        private readonly HashSet<string> _whitelist;
+        private readonly ILogger<IpWhitelistFilter> _logger;
+
+        public IpWhitelistFilter(ILogger<IpWhitelistFilter> logger, string ipWhitelistVariableName)
+        {
+            _logger = logger;
+
+            var whitelist = Environment.GetEnvironmentVariable(ipWhitelistVariableName)
+                ?? throw new EnvironmentVariableNullException(ipWhitelistVariableName);
+
+            var ips = whitelist.Split(';');
+            _whitelist = new HashSet<string>(ips); // Note: Env variable must not have spaces after semi-colons
```

Comment:
HashSet seems overkill for something we expect to have only a couple value of :thinking:

Reply:
Plus, we can make our code safe by merely checking the last character of the env var, and trimming it if its a semicolon.

Reply:
@humulla Any particular reason why this was a hashset?

Reply:
How many values do we have? In my googling, it seems like a list is more efficient up until 20 items, when a hashset becomes more useful (See the first answer [here](https://stackoverflow.com/questions/150750/hashset-vs-list-performance))

Reply:
Changed type to list